src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -541,15 +583,14 @@ void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, JavaThread* current)
         // be visible <= the ST performed by the CAS.
         lock->set_displaced_header(mark);
         if (mark == obj()->cas_set_mark(markWord::from_pointer(lock), mark)) {
-          return;
+          return true;
         }
-        // Fall through to inflate() ...
```

Comment:
You removed this comment here, but there is also this comment:
L578: // All other paths fall-through to inflate-enter.

For both of those, I would have updated to say something about
falling through to returning false instead of deletion.